File: FrontEnd/planchella-app/src/app/models/event-data.ts
import { EventSize, EventType } from './Enums';
import { EventAttachment } from './event-attachment';
import { EventAuthorData } from './event-author-data';


// export interface EventData {
//   id?: string,
//   eventType: EventType,
//   eventSize: EventSize,
//   authorData: EventAuthorData,
//   title: string,
//   description: string,
//   creationDate: Date | string | number,
//   upVotesCount?: number,
//   downVotesCount?: number,
//   eventStartDate: Date | string | number,
//   eventEndDate: Date | string | number,
//   attachments?: EventAttachment[]
// }

export interface EventData {
  id?: number,
  eventType: EventType,
  eventSize: EventSize,
  authorId: number,
  communityId: number,
  title: string,
  description: string,
  creationDate: Date | string | number,
  upvoteCount?: number,
  downvoteCount?: number,
  eventStartDate: Date | string | number,
  eventEndDate: Date | string | number,
  isUpvoted?: boolean,
  isDownVoted?: boolean,
  isStarred?: boolean,
  attachments?: EventAttachment[]
}

File: Backend/src/main/java/com/planchella/models/EventData.java
package com.planchella.models;

import com.planchella.enums.EventSize;
import com.planchella.enums.EventType;

import java.util.Date;

public class EventData {
    public EventType eventType;
    public EventSize eventSize;
    public EventAuthorData authorData;
    public String title;
    public String description ;
    public String creationDate;

    public EventData(EventType eventType, EventSize eventSize, EventAuthorData authorData,
                     String title, String description,
                     String creationDate) {
        this.eventType = eventType;
        this.eventSize = eventSize;
        this.title = title;
        this.authorData = authorData;
        this.description = description;
        this.creationDate = creationDate;
    }
    public static EventData getMockData(){
        return new EventData(EventType.HACKATHON, EventSize.LARGE, new EventAuthorData("", "Nour", ""), "Mock", "description", new Date().toString());
    }
}
